source/state_representation/test/tests/space/cartesian/test_cartesian_state.cpp

```diff
@@ -520,6 +520,93 @@ TEST(CartesianStateTest, Truthiness) {
   EXPECT_TRUE(empty);
 }
 
+TEST(CartesianStateTest, TestMultiplicationOperators) {
+  CartesianState state = CartesianState::Random("world");
+  CartesianPose pose = CartesianPose::Random("world");
+  CartesianTwist twist = CartesianTwist::Random("world");
+  CartesianAcceleration acc = CartesianAcceleration::Random("world");
+  CartesianWrench wrench = CartesianWrench::Random("world");
+
+  // CartesianState multiplied with any derived stays a CartesianState
+  auto r1 = state * state;
+  EXPECT_TRUE(r1.get_type() == StateType::CARTESIAN_STATE);
```

Comment:
nitpick: any test case that does `EXPECT_TRUE(a == b)` can be reduced to `EXPECT_EQ(a, b)`. The benefit of the latter is more informative output on test failure.
```suggestion
  EXPECT_EQ(r1.get_type(), StateType::CARTESIAN_STATE);
```